Update EIP0004 with more details (#19)
emqx/eip: active/0004-async-mnesia-change-log-replication.md

]
```

Note: transactions running on different nodes in the cluster can be recorded to the rlog table out-of-order.
Therefore traversing the rlog table twice can lead to different results.

* Non-clustered nodes fetch change logs from cluster.

> Nodes outside of the Mnesia can make use of `gen_rpc` to fetch changes from

Reviewer: ```suggestion
Nodes outside of the Mnesia cluster can make use of `gen_rpc` to fetch changes from
```

the Mnesia cluster nodes.

There are two possible models of interaction between core and replicant:

- Push model:

  Replicant nodes issue a `watch` call to one of the core nodes.

>   The core node creates a proxy process that issues `gen_rpc` calls to the replicant nodes using data about transactions that were recorded to the rlog table.

Reviewer: maybe use the word `agent` as what's used in the code ?
```suggestion
  The core node creates an agent process that issues `gen_rpc` calls to the replicant nodes using data about transactions that were recorded to the rlog table.
```

>   Once the replicant node reaches the end of the rlog table, it subscribes to mnesia events to the tables in the shard.

Author: @zmstone Please verify that this is what you envisioned.

>   Once the replicant node reaches the end of the rlog table, it subscribes to mnesia events to the tables in the shard.

Reviewer: ```suggestion
  Once the replication is close to the end of the rlog table, the agent process subscribes to mnesia events to the rlog table and start feeding the replicant with realtime stream of OPs. 
  The time threshold to identify 'close to the end of rlog' should be configurable, and realtime stream should start after (with maybe a bit overlapping) the agent reaches the `$end_of_table`
```
